src/zenml/integrations/bentoml/services/bentoml_deployment.py

```diff
@@ -0,0 +1,224 @@
+        production: whether to run in production mode
+        working_dir: working directory for the service
+        host: host to expose the service on
+    """
+
+    model_name: str
+    model_uri: str
+    bento: str
+    bento_uri: Optional[str] = None
+    apis: List[str] = []
+    workers: Optional[int] = None
+    port: Optional[int] = None
+    backlog: Optional[int] = None
+    production: bool = False
+    working_dir: str
+    host: Optional[str] = None
+
+
+class BentoMLDeploymentService(LocalDaemonService):
+    """BentoML deployment service used to start a local prediction server for BentoML models.
+
+    Attributes:
+        SERVICE_TYPE: a service type descriptor with information describing
+            the BentoML deployment service class
+        config: service configuration
+        endpoint: optional service endpoint
+    """
+
+    SERVICE_TYPE = ServiceType(
+        name="bentoml-deployment",
+        type="model-serving",
+        flavor="bentoml",
+        description="BentoML prediction service",
+    )
+
+    config: BentoMLDeploymentConfig
+    endpoint: BentoMLDeploymentEndpoint
+
+    def __init__(
+        self,
+        config: Union[BentoMLDeploymentConfig, Dict[str, Any]],
+        **attrs: Any,
+    ) -> None:
+        """Initialize the BentoML deployment service.
+
+        Args:
+            config: service configuration
+            attrs: additional attributes to set on the service
+        """
+        # ensure that the endpoint is created before the service is initialized
+        # TODO [ENG-700]: implement a service factory or builder for BentoML
+        #   deployment services
+        if (
+            isinstance(config, BentoMLDeploymentConfig)
+            and "endpoint" not in attrs
+        ):
+
+            endpoint = BentoMLDeploymentEndpoint(
+                config=BentoMLDeploymentEndpointConfig(
+                    protocol=ServiceEndpointProtocol.HTTP,
+                    port=config.port,
+                    ip_address=config.host or DEFAULT_LOCAL_SERVICE_IP_ADDRESS,
+                    prediction_url_path=BENTOML_PREDICTION_URL_PATH,
+                ),
+                monitor=HTTPEndpointHealthMonitor(
+                    config=HTTPEndpointHealthMonitorConfig(
+                        healthcheck_uri_path=BENTOML_HEALTHCHECK_URL_PATH,
+                    )
+                ),
+            )
+            attrs["endpoint"] = endpoint
+        super().__init__(config=config, **attrs)
+
+    def run(self) -> None:
+        """Start the service."""
+        logger.info(
+            "Starting BentoML prediction service as blocking "
+            "process... press CTRL+C once to stop it."
+        )
+
+        self.endpoint.prepare_for_start()
+
+        if self.config.production:
+            logger.info("Running in production mode.")
+            from bentoml.serve import serve_http_production
+
+            try:
+                serve_http_production(
```

Comment:
Is it possible to also allow user to pass in their SSL config here? 
Worth noting in docs that they can also configure BentoML home via BENTOML_HOME envar.

Reply:
Yes sure, will add SSL config